static/app/components/idBadge/teamBadge.tsx

```diff
@@ -0,0 +1,34 @@
+import TeamStore from 'sentry/stores/teamStore';
+import {useLegacyStore} from 'sentry/stores/useLegacyStore';
+import type {Team} from 'sentry/types';
+
+import BadgeDisplayName from './badgeDisplayName';
+import BaseBadge, {type BaseBadgeProps} from './baseBadge';
+
+export interface TeamBadgeProps
+  extends Partial<Omit<BaseBadgeProps, 'project' | 'organization' | 'team'>> {
+  team: Team;
+  /**
+   * When true will default max-width, or specify one as a string
+   */
+  hideOverflow?: boolean | string;
+}
+
+function TeamBadge({hideOverflow = true, team, ...props}: TeamBadgeProps) {
+  const {teams} = useLegacyStore(TeamStore);
+
+  // Get the most up-to-date team from the store
+  const resolvedTeam = teams.find(t => t.id === team.id) ?? team;
```

Comment:
if `teams` is of type `Team` there's no point in all this since we have `team: Team` passed in

Reply:
yeah idk, I didn't want to touch this part